build/cmake/GeneratePkgConfig.cmake

```diff
@@ -0,0 +1,298 @@
+    # Escape '>'
+    string(REPLACE ">" "$<ANGLE-R>" str "${in}")
+    # Escape "$"
+    string(REPLACE "$" "$<1:$>" str "${str}")
+    # Undo the escaping of the dollar for $<ANGLE-R>
+    string(REPLACE "$<1:$><ANGLE-R>" "$<ANGLE-R>" str "${str}")
+    # Escape ","
+    string(REPLACE "," "$<COMMA>" str "${str}")
+    # Escape ";"
+    string(REPLACE ";" "$<SEMICOLON>" str "${str}")
+    set("${out}" "${str}" PARENT_SCOPE)
+endfunction()
+
+# Create a generator expression that ensures the given input generator expression
+# is evaluated within the context of the named target.
+function(_bind_genex_to_target out target genex)
+    _genex_escape(escaped "${genex}")
+    set("${out}" $<TARGET_GENEX_EVAL:${target},${escaped}> PARENT_SCOPE)
+endfunction()
+
+#[==[
+Generate a pkg-config .pc file for the Given CMake target, and optionally a
+rule to install it::
+
+    generate_pkg_config(
+        <target>
+        [FILENAME <filename>]
+        [LIBDIR <libdir>]
+        [INSTALL [DESTINATION <dir>] [RENAME <filename>]]
+        [CONDITION <cond>]
+    )
+
+The `<target>` must name an existing target. The following options are accepted:
+
+FILENAME <filename>
+    - The generated .pc file will have the given `<filename>`. This name *must*
+      be only the filename, and not a qualified path. If omitted, the default
+      filename is generated based on the target name. If using a multi-config
+      generator, the default filename will include the name of the configuration
+      for which it was generated.
+
+LIBDIR <libdir>
+    - Specify the subdirectory of the install prefix in which the target binary
+      will live. If unspecified, uses `CMAKE_INSTALL_LIBDIR`, which comes from
+      the GNUInstallDirs module, which has a default of `lib`.
+
+INSTALL [DESTINATION <dir>] [RENAME <filename>]
+    - Generate a rule to install the generated pkg-config file. This is better
+      than using a `file(INSTALL)` on the generated file directly, since it
+      ensures that the installed .pc file will have the correct install prefix
+      value. The following additional arguments are also accepted:
+
+      DESTINATION <dir>
+        - If provided, specify the *directory* (relative to the install-prefix)
+          in which the generated file will be installed. If unspecfied, the default
+          destination is `<libdir>/pkgconfig`
+
+      RENAME <filename>
+        - If provided, set the filename of the installed pkg-config file. If
+          unspecified, the top-level `<filename>` will be used. (Note that the
+          default top-level `<filename>` will include the configuration type
+          when built/installed using a multi-config generator!)
+
+CONDITION <cond>
+    - The file will only be generated/installed if the condition `<cond>`
+      results in the string "1" after evaluating generator expressions.
+
+All named parameters accept generator expressions.
+
+]==]
+function(mongo_generate_pkg_config target)
```

Comment:
It baffles me that CMake doesn't provide any utilities to support smart target-based pkg-config file generation. 😞